### ISSUE TITLE
Define application structure 

### ISSUE THREAD

**Issue by Tapsi** [Tapsi/cwtactics2#13]
The structure of the application repository should be defined with this issue. 

## Points of interest

- How many source files do we have ?
- What is the release procedure ?
- How do we generate a release ?
- Unit testing

**Comment by Tapsi** [comment]
@ctomni231 feel free to add additional statements :)

**Comment by ctomni231** [comment]
How many source files do we have ?

- We should have source files for every major division in the project. Divisions are music, sound, graphics, logic, etc. These can be further split down into groups, but the rule of thumb is, if it can be modularized, then we should try to accomplish that wherever possible.

What is the release procedure ?

- Releases are dependent on milestones, which are further dependent on bullet points. When all bullet points of a Milestone is complete, a release would be made to accommodate it. The only exception to the rule is if a release needs an emergency fix or a patch update. Those can be released the moment the fix is made.

How do we generate a release ?

Release can be generated by posting an update to the gh-pages. Since our project lives online, I feel as though this is the most transparent and most direct way to push an update to the system. If people do need to test, we can have a separate package that will be available here. 

So in short, releases = gh-pages, and test packages = this Github page 

- Unit testing

Unit testing will be done at the discretion of the owner who wants to write them. Regression tests are highly recommended but not mandatory. In short, each coder is responsible for whichever unit tests they believe is necessary for program stability.
